sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1710,51 +1735,44 @@ public void startBundle(StartBundleContext c) throws IOException {
       public void processElement(@Element PubsubMessage message, @Timestamp Instant timestamp)
           throws IOException, SizeLimitExceededException {
         // Validate again here just as a sanity check.
+        // TODO(sjvanrossum): https://github.com/apache/beam/issues/31800
```

Comment:
Add a short summary of issue so it's easier to eyeball in code here

Reply:
```suggestion
        // TODO(sjvanrossum): https://github.com/apache/beam/issues/31800
        // - Size validation makes no distinction between JSON and Protobuf encoding
        // - Accounting for HTTP to gRPC transcoding is non-trivial
```